crates/compiler/can/src/constraint.rs

```diff
@@ -673,10 +676,19 @@ impl Constraints {
 
         Constraint::CheckCycle(cycle_index, cycle_mark)
     }
+
+    pub fn ingested_file(
+        &mut self,
+        type_index: TypeOrVar,
+        file_path: Box<Path>,
+        bytes: Arc<Vec<u8>>,
+    ) -> Constraint {
+        Constraint::IngestedFile(type_index, file_path, bytes)
+    }
 }
 
-roc_error_macros::assert_sizeof_default!(Constraint, 3 * 8);
-roc_error_macros::assert_sizeof_aarch64!(Constraint, 3 * 8);
+roc_error_macros::assert_sizeof_default!(Constraint, 4 * 8);
+roc_error_macros::assert_sizeof_aarch64!(Constraint, 4 * 8);
```

Comment:
this is unfortunate. there is no way around it?

Reply:
Found a fix. Using `Box<PathBuf>` (yes this is more indirection), to avoid `Box<Path>` turning into a fat pointer and costing 16 bytes. I think the indirection here is worth it to avoid the costs of growing these types. The data here is really only used in limited places.

Reply:
PathBuf is already boxed, any reason to put it in a Box again?

Reply:
underlying PathBuf is a `Vec<u8>` so it is 24 bytes, not 8.